test/extensions/secondorder/diag_ggn/test_diag_ggn.py

```diff
@@ -0,0 +1,69 @@
+from test.automated_test import check_sizes_and_values
+from test.extensions.problem import make_test_problems
+from test.extensions.implementation.autograd import AutogradExtensions
+from test.extensions.implementation.backpack import BackpackExtensions
+from test.extensions.secondorder.diag_ggn.diaggnn_settings import DiagGGN_SETTINGS
+
+import pytest
+import torch
+
+
+PROBLEMS = make_test_problems(DiagGGN_SETTINGS)
+IDS = [problem.make_id() for problem in PROBLEMS]
+
+
+@pytest.mark.parametrize("problem", PROBLEMS, ids=IDS)
+def test_diag_ggn(problem):
+    """Test the diagonal of Gauss-Newton
+
+    Args:
+        problem (ExtensionsTestProblem): Problem for extension test.
+    """
+    problem.set_up()
+
+    backpack_res = BackpackExtensions(problem).diag_ggn()
+    autograd_res = AutogradExtensions(problem).diag_ggn()
+
+    check_sizes_and_values(autograd_res, backpack_res)
+    problem.tear_down()
+
+
+@pytest.mark.parametrize("problem", PROBLEMS, ids=IDS)
+def test_diag_ggn_mc_light(problem):
+    """Test the MC approximation of Diagonal of Gauss-Newton
+        with few mc_samples (light version)
+
+    Args:
+        problem (ExtensionsTestProblem): Problem for extension test.
+    """
+    problem.set_up()
+    torch.manual_seed(0)
```

Comment:
`problem.set_up` already sets a seed. Please check, and if so, remove this line.